ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -403,201 +503,147 @@ func extractPod(admissionReview *admissionv1.AdmissionReview) (*corev1.Pod, erro
-					workerID := corev1.EnvVar{
-						Name:  "TPU_WORKER_ID",
-						Value: fmt.Sprint(tpuWorkerID),
-					}
-					idPatch := patch{"op": "add"}
-					// create new EnvVar array if container.Env is empty, and append new EnvVars if not
-					if len(container.Env) == 0 {
-						idPatch["path"] = path
-						idPatch["value"] = []corev1.EnvVar{workerID}
-					} else {
-						idPatch["path"] = fmt.Sprintf("%s/-", path)
-						idPatch["value"] = workerID
-					}
-					patches = append(patches, idPatch)
-				}
-				// inject TPU_NAME
-				if getEnvironmentVariable("TPU_NAME", container) == "" {
-					tpuNameValue := fmt.Sprintf("%s-%d", groupName, replicaIndex)
-					klog.V(1).InfoS("mutatePod", "RayCluster", namespace+"/"+clusterName, "TPU_NAME", tpuNameValue, "Replica Index", replicaIndex)
-					tpuName := corev1.EnvVar{
-						Name:  "TPU_NAME",
-						Value: tpuNameValue,
-					}
-					namePatch := patch{"op": "add"}
-					// create new EnvVar array if container.Env is empty, and append new EnvVars if not
-					if len(container.Env) == 0 {
-						namePatch["path"] = path
-						namePatch["value"] = []corev1.EnvVar{tpuName}
-					} else {
-						namePatch["path"] = fmt.Sprintf("%s/-", path)
-						namePatch["value"] = tpuName
-					}
-					patches = append(patches, namePatch)
-				}
-			}
-		}
-	}
-
-	patchBytes, err := json.Marshal(patches)
-	if err != nil {
-		return nil, err
-	}
-
 	admissionResponse := &admissionv1.AdmissionResponse{
 		UID:     admissionReview.Request.UID,
 		Allowed: true,
-		Patch:   patchBytes,
-		PatchType: func() *admissionv1.PatchType {
-			pt := admissionv1.PatchTypeJSONPatch
-			return &pt
-		}(),
 	}
-	return admissionResponse, nil
-}
 
-// update sliceToWorkers map on pod deletion
-func deletePod(admissionReview *admissionv1.AdmissionReview) (*admissionv1.AdmissionResponse, error) {
-	// Create AdmissionResponse - we never deny the deletion request
-	admissionResponse := &admissionv1.AdmissionResponse{
-		UID:     admissionReview.Request.UID,
-		Allowed: true,
-		Result: &metav1.Status{
-			Status:  "Success",
-			Message: "",
-		},
+	containers := pod.Spec.Containers
+	if containers == nil {
+		return nil, errors.New("Container path not specified")
 	}
-
-	pod, err := extractPod(admissionReview)
-	if err != nil {
-		klog.Errorf("Pod extraction failed: %s", err)
+	if !containerRequestingTPUs(containers...) {
+		// if no TPUs are requested, simply admit the Pod
+		return admissionResponse, nil
 	}
 
+	// ray operator only sets GenerateName field - doesn't include random suffix until after admission request
+	// use mapping of {cluster name, group name, replicaIndex} -> workers to extract next TPU_WORKER_ID
 	clusterName := pod.Labels["ray.io/cluster"]
 	if clusterName == "" {
-		return admissionResponse, errors.New("Kuberay Pod missing RayCluster label")
+		return nil, errors.New("KubeRay Pod missing RayCluster label")
```

Comment:
KubeRay Pod is misleading. Maybe just `Ray Pod`?

Reply:
Or to be very specific, `Ray Pod created by KubeRay missing label `ray.io/cluster`

Reply:
Done in [869f275](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/869f275fb540b413bd9c916b921d3e94b0dee197)